server.js

```diff
@@ -21,9 +15,116 @@ app.use(express.json());
 
 // Start defining your routes here
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  const endpoints = listEndpoints(app)
+  res.json(endpoints);
+});
+
+// Shark attacks endpoint to receive a collection of all shark attacks
+app.get("/shark-attacks", (req, res) => {
+  res.json(globalSharkAttackData)
 });
 
+// Shark attacks "id" endpoint, (used the original_order in json), to receive a single shark attack
+app.get("/shark-attacks/:id", (req, res) => {
+  const {id} = req.params
+  const sharkAttack = globalSharkAttackData.find(sharkAttack => sharkAttack.original_order === +id)
+
+  if (sharkAttack) {
+    res.json(sharkAttack)
+  } else {
+  res.status(404).send("Unfortunately for you, but luckily for the swimmers - No shark attack was found!")
+}
+})
+
+// Year of shark attacks endpoint. Also added query param for the possibility to filter if the shark attacks from a certain year was unprovoked
+app.get("/shark-attacks/year/:year", (req, res) => {
+  const year = req.params.year
+  const typeUnprovoked = req.query.type
+  let sharkAttacksFromYear = globalSharkAttackData.filter((sharkAttack) => sharkAttack.Year === +year)
+
+if (sharkAttacksFromYear.length === 0) {
+  return res.status(404).json({ error: "Calm waters - No shark attacks found from this year!" })
+}
+
+if (typeUnprovoked) {
+sharkAttacksFromYear = sharkAttacksFromYear.filter((sharkAttack) => sharkAttack.Type.toLowerCase() === typeUnprovoked.toLowerCase())
+}
```

Comment:
I'm just thinking that maybe this param in the if could be named "type" only? I was first a bit confused since I thought the query could only be unprovoked 😊 It's superclear in the read me, so the confusion from my end didn't appear until I read the code. Just a small thought 🧡